File: Springboot Code/AgroRent/src/main/java/com/majorproj/agrorent/dto/BookingReqDto.java
 package com.majorproj.agrorent.dto;

import java.time.LocalDate;

import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class BookingReqDto {
	@NotNull(message = "Start date is required")
    @Future(message = "Start date must be in the future")
    private LocalDate startDate;

    @NotNull(message = "End date is required")
    @Future(message = "End date must be in the future")
    private LocalDate endDate;

    @NotNull(message = "Total amount is required")
    @Positive(message = "Total amount must be positive")
    private Double totalAmount;

    @NotNull(message = "Equipment ID is required")
    private Long equipmentId;

}

File: Dot Net Code/AgroRent/DTOs/BookingReqDto.cs
using System.ComponentModel.DataAnnotations;

namespace AgroRent.DTOs
{
    public class BookingReqDto
    {
        [Required]
        public int EquipmentId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }
    }
}
